### ISSUE TITLE
VPN-Container RDP issues after updating docker packages

### ISSUE THREAD

**Issue by hakimdotdev** [moby/moby#47326]
### Description

We use ubuntu containers as clients for various VPN-Clients for RDP-Sessions. Since an update of the apt package, the connection is very unstable. The VPN Connection is established, and remote servers and the port for RDP are reachable but it will loose connection sporadically. The only thing that has changed is the version of the packages mentioned below.

### Reproduce

On Debian 12, install the following packages:

```
containerd.io/bookworm 1.6.27-1 amd64 [upgrade from 1.6.26-1].
docker-buildx-plugin/bookworm 0.12.1-1~debian.12~bookworm amd64 [upgraded from 0.11.2-1~debian.12~bookworm].
docker-ce-cli/bookworm 5.25.0.1-1~debian.12~bookworm amd64 [upgrade from 5.24.0.7-1~debian.12~bookworm].
docker-ce-rootless-extras/bookworm 5.25.0.1-1~debian.12~bookworm amd64 [upgrade from 5.24.0.7-1~debian.12~bookworm].
docker-ce/bookworm 5.25.0.1-1~debian.12~bookworm amd64 [upgrade from 5.24.0.7-1~debian.12~bookworm].
docker-compose-plugin/bookworm 2.24.2-1~debian.12~bookworm amd64 [upgrade from 2.21.0-1~debian.12~bookworm].
```
The _upgrade from_ versions are the last working ones.

Set up a ubuntu:latest container and a vpn connection:

```dockerfile
FROM ubuntu:latest

# Set the timezone
ENV TZ="Europe/Berlin

# Install Linux packages

RUN apt-get update -y \
    && DEBIAN_FRONTEND=non-interactive apt-get upgrade -y \
    && DEBIAN_FRONTEND=non-interactive apt-get install -y --no-install-recommends \
    ca-certificates \
    tzdata \
    expect \
    net-tools \
    openssh-client \
    iproute2 \
    iptables \
    wget \
    iputils-ping \ 
    socat \
    netcat \
    openvpn \
    tini \
    && apt-get clean \
    && apt-get autoremove --purge \
    \ && rm -rf /var/lib/apt/lists/*
```

Expose private port 3389 and have container forward all traffic from mapped public port to private IP (remote Windows server). This will cause massive dropped connections when opening an RDP session from a client.

### Expected behavior

Stable RDP-Sessions

### docker version

```bash
Client: Docker Engine - Community
 Version:           24.0.7
 API version:       1.43
 Go version:        go1.20.10
 Git commit:        afdd53b
 Built:             Thu Oct 26 09:08:02 2023
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          24.0.7
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.10
  Git commit:       311b9ff
  Built:            Thu Oct 26 09:08:02 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.26
  GitCommit:        3dd1e886e55dd695541fdcd67420c2888645a495
 runc:
  Version:          1.1.10
  GitCommit:        v1.1.10-0-g18a0cb0
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client: Docker Engine - Community
 Version:    24.0.7
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.21.0
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 82
  Running: 25
  Paused: 0
  Stopped: 57
 Images: 27
 Server Version: 24.0.7
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dd1e886e55dd695541fdcd67420c2888645a495
 runc version: v1.1.10-0-g18a0cb0
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.1.0-13-amd64
 Operating System: Debian GNU/Linux 12 (bookworm)
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 15.58GiB
 Name: <thehostname>
 ID: 2870ce1d-4882-4ff7-b2e7-f4df32fabddf
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Comment by akerouanton** [comment]
Hi @hakimdotdev, I was going to answer your report but you just closed it. Could you confirm this was an issue on your side (eg. faulty network, etc...)?

**Comment by hakimdotdev** [comment]
It wasn't, i just thought its the wrong repo to post this.

**Comment by thaJeztah** [comment]
What version of docker were you running before updating?

**Comment by thaJeztah** [comment]
Oh! The `docker info` output is from _before_ updating, not the version you ran into issues with, is that correct?

```
docker-ce-cli/bookworm 5.25.0.1-1~debian.12~bookworm amd64 [upgrade from 5.24.0.7-1~debian.12~bookworm].
```

**Comment by hakimdotdev** [comment]
@thaJeztah Yes you're right, i can't reproduce right now, as the replica system now got a even newer version via auto updates. I'll have to reinstall the mentioned versions and get a `version` and `info output`

**Comment by thaJeztah** [comment]
Thanks! Sorry for adding some noise; I saw the `24.0.7` in the `docker info` / `docker version`, and got curious because I didn't recall any significant changes in patch releases for 24.0.x in networking, but then I spotted your comment that it was when upgrading _to_ 25.0.x (from 24.0.x) 😅 .

**Comment by akerouanton** [comment]
Could you confirm you updated straight from 24.0.7 to 25.0.1, without going through 25.0.0? Your report makes me think it could be a duplicate of https://github.com/moby/moby/issues/47211. It's worth checking you don't have any duplicate MAC address in your docker networks.

If it's inconclusive, could you try running `ping` continuously from both your containers and your host (targetting the remote system you usually connect to) until you see the sporadic issue re-happen again? If you observe packet losses only from the container, it'd rule out any external factors (eg. flaky link, etc...).

Could you enable debug logs (see here: https://docs.docker.com/config/daemon/logs/) and check if the issue is correlated to any particular API calls made to the Engine?

Finally, could you check with tcpdump what really happens (eg. packets dropped, ICMP error code returned, TCP RST, etc...)?

**Comment by hakimdotdev** [comment]
Hi @akerouanton, 

> Could you confirm you updated straight from 24.0.7 to 25.0.1, without going through 25.0.0? Your report makes me think it could be a duplicate of https://github.com/moby/moby/issues/47211. It's worth checking you don't have any duplicate MAC address in your docker networks.

I'm pretty sure but i can't confirm it right now, in the System /w old versions, i can verify it but I'll have to spin up the staging environment to test it.

> If it's inconclusive, could you try running ping continuously from both your containers and your host (targetting the remote system you usually connect to) until you see the sporadic issue re-happen again? If you observe packet losses only from the container, it'd rule out any external factors (eg. flaky link, etc...).

I've tried this when the issue occured. There is no packet losses, IPs and Ports are reachable when pinging from container to remote server while RDP says re-establishing.

> Could you enable debug logs (see here: https://docs.docker.com/config/daemon/logs/) and check if the issue is correlated to any particular API calls made to the Engine?

> Finally, could you check with tcpdump what really happens (eg. packets dropped, ICMP error code returned, TCP RST, etc...)?

I'll spin up the staging env and try that.

**Comment by hakimdotdev** [comment]
I've looked at some other Issues.

The System has a nginx on it and Docker runs behind it ([grafana#81258](https://github.com/grafana/grafana/issues/81258)) and all VPN-Containers are on a created bridged Network #47211

**Comment by thaJeztah** [comment]
@aitchdot I see you closed this one; was it resolved on your side? (if so 🎉) Was there more context to share in case others would run into something similar?
